Update documentation for `lufact!`. Fixes #16193.

gasagna opened the pull request:
Reference JuliaLang/LinearAlgebra.jl#327

gasagna committed:
Updated documentation for lufact!, qrfact!, cholfact! and scale!

Update documentation for `lufact!`. Fixes #16193.

removed documentation for sparse lufact

improved documentation for qrfact

improved documentation for scale

improved documentation for cholfact!

Revert changes to .rst

This reverts commit 9f727c475456da6749be3c94d928a1062d42ee9a.

Improved documentation of cholfact!

Removed trailing whitespace

run make docs to apply changes to .rst file

Changed double to single backticks

updated linalg.rst

andreasnoack commented:
Great. Would you mind adding similar sentences to `cholfact!` and `qrfact!`? Maybe also `scale!`.

tkelman commented on base/docs/helpdb/Base.jl:
```diff
@@ -5671,10 +5671,12 @@ Base.(:(.!=))
+`lufact!` is the same as [`lufact`](:func:`lufact`), but saves space by overwriting the 
+input `A`, instead of creating a copy. An `InexactError` exception is thrown if the 
+factorisation produces a number not representable by the element type of `A`, e.g. for 
+integer types. For sparse `A` the `nzval` field is not overwritten but the index fields, 
```
@andreasnoack didn't we get rid of sparse `lufact!` for now?

andreasnoack replied:
Yes. There is no sparse `lufact!`.

gasagna commented:
Ok. Removed trailing whitespace, added similar sentences to qrfact! and cholfact!, ran make docs and committed changes to .rst.  Should I do something else?

tkelman commented on base/linalg/cholesky.jl:
```diff
@@ -126,7 +126,10 @@ end
-The same as `cholfact`, but saves space by overwriting the input `A`, instead of creating a copy.
+The same as `cholfact`, but saves space by overwriting the input `A`, instead
+of creating a copy. An ``InexactError`` exception is thrown if the factorisation
+produces a number not representable by the element type of ``A``, e.g. for
```
single backticks in markdown unless you want latex style output

tkelman commented on base/linalg/cholesky.jl:
```diff
@@ -139,7 +142,10 @@ end
 
-The same as `cholfact`, but saves space by overwriting the input `A`, instead of creating a copy.
+The same as `cholfact`, but saves space by overwriting the input `A`, instead
+of creating a copy. An ``InexactError`` exception is thrown if the
```
single backticks

gasagna commented:
WOW, just gone through my first rebase. Should be ok now. Thanks for reviewing.

andreasnoack commented:
LGTM. @tkelman Do you see anything that requires adjustments?
